functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPomXml.java

```diff
@@ -21,68 +24,91 @@
+import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.Test;
 
 public class BayesianPomXml {
 
   private static final Logger LOG = LoggerFactory.getLogger(TestTestClass.class);
 
-  @Inject private RhCheTestWorkspaceImpl workspace;
+  @InjectTestWorkspace(template = RhCheWorkspaceTemplate.RH_DEFAULT)
+  private TestWorkspace workspace;
+
   @Inject private NavigateToFile navigateToFile;
   @Inject private Loader loader;
   @Inject private CodenvyEditor editor;
   @Inject private Ide ide;
   @Inject private NotificationsPopupPanel notificationsPopupPanel;
   @Inject private ProjectExplorer projectExplorer;
 
-  private static final Integer pomExpectedErrorLine = 40;
-  private static final Integer pomInjectionEntryPoint = 37;
-  private static final String POM_FILE = "pom.xml";
+  private static final Integer POM_EXPECTED_ERROR_LINE = 40;
+  private static final Integer POM_INJECTION_ENTRY_POINT = 37;
+  private static final String PROJECT_FILE = "pom.xml";
   private static final String ERROR_MESSAGE =
       "Package ch.qos.logback:logback-core-1.1.10 is vulnerable: CVE-2017-5929";
   private static final String PROJECT_NAME = "vertx-http-booster";
-  private static final String pomDependency =
+  private static final String PROJECT_DEPENDENCY =
       "<dependency>\n"
           + "<groupId>ch.qos.logback</groupId>\n"
           + "<artifactId>logback-core</artifactId>\n"
           + "<version>1.1.10</version>\n"
           + "</dependency>\n";
 
-  @Test(priority = 1)
-  public void openClass() throws Exception {
-    ide.open(workspace);
-
-    // this block of code ensures that workspace and project is ready and pop-ups are gone
-    // still doesn't seem to be
-    ide.waitOpenedWorkspaceIsReadyToUse();
-    projectExplorer.waitProjectExplorer();
-    projectExplorer.waitItem(PROJECT_NAME);
-    notificationsPopupPanel.waitProgressPopupPanelClose();
+  @BeforeClass
+  public void openTestFile() throws Exception {
+    try {
+      LOG.info(
+          "Workspace with name: "
+              + workspace.getName()
+              + " and id: "
+              + workspace.getId()
+              + " was successfully injected. ");
+      ide.open(workspace);
+      ide.waitOpenedWorkspaceIsReadyToUse();
+      projectExplorer.waitProjectExplorer();
+      notificationsPopupPanel.waitProgressPopupPanelClose();
+      projectExplorer.waitItem(PROJECT_NAME);
+    } catch (ExecutionException | InterruptedException e) {
+      LOG.error(
+          "Could not obtain workspace name and id - worskape was probably not successfully injected.");
+      throw e;
+    } catch (Exception e) {
+      LOG.error("Could not open workspace IDE.");
+      throw e;
+    }
+  }
 
+  @BeforeMethod
+  public void prepareProjectFile() {
     openDefinedClass();
+    appendDependency();
+  }
+
+  @AfterMethod
+  public void closeFiles() {
+    removeDependency();
+    editor.closeAllTabs();
   }
 
-  @Test(priority = 2)
+  @Test
```

Comment:
Why did you remove priority from here?

Reply:
Both tests were re-written to be atomic, the order therefore doesn't matter